src/modules/ethereum/message-signing/signTypedData.test.ts

```diff
@@ -0,0 +1,65 @@
+import { ethers } from 'ethers';
+import type { LocallyEncoded } from 'src/shared/wallet/encode-locally';
+import { decodeMasked } from 'src/shared/wallet/encode-locally';
+import { opaqueType } from 'src/shared/type-utils/Opaque';
+import { signTypedData } from './signTypedData';
+
+const sample = {
+  typedData: {
+    types: {
+      Transaction: [
+        { name: 'txType', type: 'uint256' },
+        { name: 'from', type: 'uint256' },
+        { name: 'to', type: 'uint256' },
+        { name: 'gasLimit', type: 'uint256' },
+        { name: 'gasPerPubdataByteLimit', type: 'uint256' },
+        { name: 'maxFeePerGas', type: 'uint256' },
+        { name: 'maxPriorityFeePerGas', type: 'uint256' },
+        { name: 'paymaster', type: 'uint256' },
+        { name: 'nonce', type: 'uint256' },
+        { name: 'value', type: 'uint256' },
+        { name: 'data', type: 'bytes' },
+        { name: 'factoryDeps', type: 'bytes32[]' },
+        { name: 'paymasterInput', type: 'bytes' },
+      ],
+      EIP712Domain: [
+        { name: 'name', type: 'string' },
+        { name: 'version', type: 'string' },
+        { name: 'chainId', type: 'uint256' },
+      ],
+    },
+    domain: { name: 'zkSync', version: '2', chainId: '543210' },
+    primaryType: 'Transaction',
+    message: {
+      txType: 113,
+      from: '0xE093d671dA3D42fBf79BC333692a7A5f794EdDFb',
+      to: '0x6a6394f47dd0baf794808f2749c09bd4ee874e70',
```

Comment:
Not sure is it important or not, to have everywhere `from` in mixed case and `to` in lowercase?

Reply:
Hmm I think I copied objects as is. Ideally this shouldn't matter